driver/src/main/java/org/kaazing/k3po/driver/internal/netty/bootstrap/file/FileChannel.java

```diff
@@ -138,6 +138,9 @@ private static MappedByteBuffer mapFile(URI fileAddress, String mode, long size)
                 break;
             case "rw":
                 mapMode = READ_WRITE;
+                // parent directory must exist to create file
+                File parentDir = location.getParentFile();
+                parentDir.mkdirs();
```

Comment:
I think we have two options. Either to create parent dir (as above) or throw exception if the parent dir doesn't exist.

Also, do we need to guard against NPE  that may occur due to other reasons (create may fail due to permissions etc).